zaza/openstack/charm_tests/vault/utils.py

```diff
@@ -280,6 +281,65 @@ def ensure_secret_backend(client):
         pass
 
 
+def wait_until_ready(clients):
+    """Wait until vault is ready (for all clients).
+
+    Raise an error if it's not ready within a reasonable time.
+
+    :param clients: Clients to use to talk to vault
+    :type clients: List[CharmVaultClient]
+    :raises: AssertionError
+    """
+    for client in clients:
+        for attempt in tenacity.Retrying(
+            reraise=True,
+            wait=tenacity.wait_fixed(10),
+            stop=tenacity.stop_after_attempt(12),  # wait for max 2 minutes
+        ):
+            with attempt:
+                ha_status = client.hvac_client.ha_status
+                if (
+                    not ha_status.get('leader_address') and
+                    ha_status.get('ha_enabled')
+                ):
+                    raise AssertionError('vault not ready: no leader_address')
+
+
+def wait_until_all_initialised(clients):
+    """Wait until vault is initialised (for all clients).
+
+    Raise an error if it's not initialised within a reasonable time.
+
+    :param clients: Clients to use to talk to vault
+    :type clients: List[CharmVaultClient]
+    :raises: AssertionError
+    """
+    for client in clients:
+        for _ in range(12):
+            if is_initialized(client):
+                break
+            time.sleep(10)
+        else:
+            raise AssertionError("Timeout waiting for vault to initialise")
+
+
+def wait_until_one_initialised(clients):
+    """Wait until at least one vault unit is initialised.
+
+    Raise an error if it's not initialised within a reasonable time.
+
+    :param clients: Clients to use to talk to vault
+    :type clients: List[CharmVaultClient]
+    :raises: AssertionError
```

Comment:
This block needs an `:returns:` and `:rtype:` as it returns the `CharmVaultClient` that is initialised.